scalafix-tests/unit/src/test/scala/scalafix/tests/interfaces/ScalafixArgumentsSuite.scala

```diff
@@ -345,6 +345,33 @@ class ScalafixArgumentsSuite extends AnyFunSuite with DiffAssertions {
     assert(fileEvaluation.getErrorMessage.get.contains("SemanticDB not found"))
   }
 
+  test("Scala2 source with Scala3 syntax can only be parsed with -Xsource:3") {
+    val cwd = StringFS
+      .string2dir(
+        s"""|/src/Scala2Source3.scala
+          |open class Scala2Source3""".stripMargin,
+        charset
+      )
+      .toNIO
+    val src = cwd.resolve("src")
+    val args = api
+      .withScalaVersion("2.13.6")
```

Comment:
This value is not mandatory, but I left it there since what we expect clients to send - I feel like it's enough/better to test for the presence of `-Xsource:` only, so that this PR works in the future with the 2.12.x release that will eventually allow it as well.

Reply:
I think it's ok